vnext/Microsoft.ReactNative/Fabric/Composition/ReactNativeIsland.cpp

```diff
@@ -98,20 +98,38 @@ void CompositionReactViewInstance::UninitRootView() noexcept {
   }
 }
 
-void ApplyConstraints(
+void ReactNativeIsland::ApplyConstraints(
     const winrt::Microsoft::ReactNative::LayoutConstraints &layoutConstraintsIn,
-    facebook::react::LayoutConstraints &layoutConstraintsOut) noexcept {
+    facebook::react::LayoutConstraints &layoutConstraintsOut) const noexcept {
   layoutConstraintsOut.minimumSize = {layoutConstraintsIn.MinimumSize.Width, layoutConstraintsIn.MinimumSize.Height};
   layoutConstraintsOut.maximumSize = {layoutConstraintsIn.MaximumSize.Width, layoutConstraintsIn.MaximumSize.Height};
-  layoutConstraintsOut.layoutDirection =
-      static_cast<facebook::react::LayoutDirection>(layoutConstraintsIn.LayoutDirection);
+  if (layoutConstraintsIn.LayoutDirection == winrt::Microsoft::ReactNative::LayoutDirection::Undefined) {
+    if (m_island) {
+      layoutConstraintsOut.layoutDirection =
+          (m_island.LayoutDirection() == winrt::Microsoft::UI::Content::ContentLayoutDirection::LeftToRight)
+          ? facebook::react::LayoutDirection::LeftToRight
+          : facebook::react::LayoutDirection::RightToLeft;
+    } else if (m_hwnd) {
+      auto styles = GetWindowLongPtrW(m_hwnd, GWL_EXSTYLE);
+      layoutConstraintsOut.layoutDirection = ((styles & WS_EX_LAYOUTRTL) == WS_EX_LAYOUTRTL)
+          ? facebook::react::LayoutDirection::RightToLeft
+          : facebook::react::LayoutDirection::LeftToRight;
+    }
+  } else {
+    layoutConstraintsOut.layoutDirection =
+        static_cast<facebook::react::LayoutDirection>(layoutConstraintsIn.LayoutDirection);
+  }
 }
 
-ReactNativeIsland::ReactNativeIsland() noexcept {}
+ReactNativeIsland::ReactNativeIsland() noexcept {
+  InitTextScaleMultiplier();
```

Comment:
should we similarly attempt to initialize `m_scaleFactor` here, or do we not yet have our Island context we can query against and the scale gets updated later on attach?

Reply:
yeah, scale is harder to get, and we cannot init it here.  In most cases the scale will come from the ContentIsland when we hook it up.  In some more edge cases such as prerendering before hosting, or hosting without a ContentIsland the scale needs to be set manually from outside the ReactNativeIsland.